src/tiled/terrainbrush.cpp

```diff
@@ -204,7 +204,7 @@ void TerrainBrush::capture()
 
     const QPoint position = tilePosition() - tileLayer->position();
 
-    if (!tileLayer->contains(position))
+    if (!tileLayer->contains(position) && !mapDocument()->map()->infinite())
```

Comment:
This check is no longer necessary. I think capturing outside of the map can simply capture "no terrain", also for finite maps.